### PR TITLE
APC balance tweaks

### PR DESCRIPTION
**Pull request by JeserTheFox**
## About The Pull Request

1. Decreased acid and slash resistances
2. Increased explosive and blunt resistances
3. MED APC slots are 8 passengers + 6 revivable now.
4. Added foamed metal bump special case.
5. APC now deals a bit of damaged to allies in order to discourage VCs from ramming marines without urgent reason. Includes big warning for driver and adjusted the FF log for staff.
6. Slightly buffed ramming damage to hostile faction humans.
7. Queen can be pushed only from the back now.
8. Acid spray now deals damage to vehicle's locomotion module directly.
9. Acid traps now trigger on multitiles.
10. Acid spray from acid traps deal less damage to vehicles than those sprayed by xenos due to group effect.
11. Locomotion modules can be acid-resistant, which cuts damage dealt in half.
12. Xenos have separate var for melee vehicle damage
13. Increased Ravager and Queen damage to vehicles. Both of them do double damage.
14. Hull also takes a bit of damage now even if not all hardpoints are destroyed.
15. Small nerf of acid smoke damage to vehicles, due to acid resistance nerf.
16. Made SensorMates mandatory in Vehicle Crewmen vendor.
17. Hull also takes a bit of damage now even if not all hardpoints are destroyed.
18. APC gunner gets slight zoom to increase range from which it can provide covering fire for marines without blocking them into xenos.

This may be tweaked or even seriously changed depending on testmerge results.

https://user-images.githubusercontent.com/43144601/177216864-c12610d0-3340-4bd1-b1dc-19ca20f4b11c.mp4

изображение

## Why It's Good For The Game

These nerfs should discourage APC frontlining, due to increased damage received. Ravs made more scary to vehicles, presence of Queen, in theory, should encourage APC to leave the area and let marines handle it. Bodyblocking is still possible and valid strategy, but it's a last resort tactic, which has very high chance to result in losing modules or entire vehicle. And even if that won't happen, VCs wil lactually be forced to retreat and repair after each such maneuvor, instead of being able to pull of a dozen of them without repairs.
TLDR: constant frontlining and bodyblocking is not okay, bodyblocking to prevent marine capture or to close FOB breach is still valid tactic.
Ramming change should discourage ramming dozens of marines instead of honking and waiting 2-3 seconds first. Ramming marines should be done only during actual emergencies of if they are AFK and standing in the middle of the road. And VCs definitely don't have serious reasons to ram every marine they meet on Almayer pre-drop.
Acid spray change adds new unique way to afflict damage to vehicle for spitters and burowers, who now can go for mobility kill.

## Changelog

:cl: Jeser

balance: APC armor received explosive and blunt damage resistances buff as well as slash and acid resistances nerf.
balance: Vehicle hull now takes little bit of damage even if there are no destroyed modules.
balance: MED APC has 8 passengers (same as CMD APC) and 6 revivable bodies slots now.
balance: APC now deals a bit of damage when ramming allies in order to discourage Vehicle Crewmen from ramming marines without serious reason like delivering near-perma patient to medics. Includes big warning for driver and adjusted FF log for staff.
balance: Queen can be pushed only from the back now.
balance: Acid spray from acid traps and some xenos abilities damages vehicle locomotion (wheels or treads) modules directly. Acid traps are trigggered by vehicles now. Some locomotion modules can be acid-resistant, which cuts damage received by half.
balance: T3s overall do more damage to vehicles than before. Queen now also deals double damage to vehicles just as Ravager.
balance: Boilers acid gas damage to vehicles was nerfed, due to overall acid resistance nerf of APC armor.
code: Made SensorMates mandatory in Vehicle Crewman vendor.

/:cl:

### CODE REVIEW

**Comment by JohnDoe50974** [comment]
Please do not make APC rams damage marines. This will only make the VC situation even worse.

**Comment by Diegoflores31** [comment]
the only thing i see with this tweak is the apc being more useless and a single squad of 3 spitters or lurkers  sindlehandledly destroying the apc completely. specially having into consideration that the apc weapons are mostly peashooters

**Comment by FoxyStalin** [comment]
'Acid spray damages the locomotion modual directly' sounds like it's going to be miserable. It sounds like a spitter or prae or two could just nuke the wheels without much risk, and if you lose the wheels you lose the APC.
It'd be different if you could purchase replacement parts, but, you cannot.

**Comment by SolaAqua** [comment]
Sounds like it will just kill the fun of being a vc and be hated by marines.I know marines love it when they take damage cause of ff,hope they love it more as apcs doing it now.I know all vcs love repairing and ya buff it’s resistances to nonxeno attacks and increase xeno attacks. I know ya want apc to back lines but it’s just going to be a sg with wheels. Only good thing I see bout the pr is boiler gas not agreeing apc,queen can block from sides and front Instead of just the  front,and I ncreased rav and queen damage but that was done awhile back rav and queen do more damage then normal xenos(know this sounds like hate but colony van is going to be more usefully then apc)

**Comment by Wintermote** [comment]
If you want the APC to stop frontlining the easiest and fastest way to achieve that is to extend the range of the guns on it so it doesn't need to frontline to shoot and kill anything, and increase the damage it takes from marine gunfire.

I know there's some dumb "but xenos shouldn't be shot at long range by long ranged marine weapons because why should they have drawbacks to being melee focused"

but that is how you're going to get the APC players to stop frontlining right in front of all the marines.

File: code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm [hunk 1]
```diff
@@ -325,6 +325,7 @@ GLOBAL_LIST_INIT(cm_vending_clothing_vehicle_crew, list(
 		list("Gloves", 0, /obj/item/clothing/gloves/yellow, MARINE_CAN_BUY_GLOVES, VENDOR_ITEM_MANDATORY),
 		list("Tanker Armor", 0, /obj/item/clothing/suit/storage/marine/tanker, MARINE_CAN_BUY_ARMOR, VENDOR_ITEM_MANDATORY),
 		list("M50 Tanker Helmet", 0, /obj/item/clothing/head/helmet/marine/tanker, MARINE_CAN_BUY_HELMET, VENDOR_ITEM_MANDATORY),
+		list("SensorMate HUD", 0, /obj/item/clothing/glasses/hud/sensor, MARINE_CAN_BUY_MRE, VENDOR_ITEM_MANDATORY),
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm:328]: `		list("SensorMate HUD", 0, /obj/item/clothing/glasses/hud/sensor, MARINE_CAN_BUY_MRE,
VENDOR_ITEM_MANDATORY),
`
this can buy MRE thing is not gonna work, swap it out for can buy glasses or something. Or can buy special.

File: code/modules/vehicles/hardpoints/wheels/locomotion.dm [hunk 2]
```diff
@@ -30,3 +31,18 @@
...
+	var/damage_dealt = acid.damage_amount
+	//First we check source of acid. Due to traps generating 3x3 acid spray field and triggering only when at least 4 tiles
+	//of vehicle enter the spray spawn area, it deals a huge amount of damage. But simply nerfing damage will also nerf it for
+	//acid spraying castes like spitters and praetorians, which is not ideal.
+	if(acid.cause_data.cause_name == "resin acid trap")
+		damage_dealt = round(damage_dealt / 3)
+	//then we check whether this locomotion module is acid-resistant
+	if(acid_resistant)
+		damage_dealt = damage_dealt / 2
+	health -= damage_dealt
```
Comment (stanalbatross) [inline code/modules/vehicles/hardpoints/wheels/locomotion.dm:46]: use take_damage here?

File: code/modules/vehicles/multitile/multitile_bump.dm [hunk 3]
```diff
@@ -469,14 +478,15 @@
 			KnockDown(0.5, 1)
 		else
 			KnockDown(1, 1)
-
 	else if(V.vehicle_flags & VEHICLE_CLASS_LIGHT)
+		dmg = TRUE
 		if(get_target_lock(driver.faction))
 			KnockDown(0.5, 1)
+			apply_damage(5 + rand(0, 5), BRUTE)
+			to_chat(V.seats[VEHICLE_DRIVER], SPAN_WARNING(SPAN_BOLD("*YOU RAMMED AN ALLY AND HURT THEM!*")))
```
Comment (stanalbatross) [inline code/modules/vehicles/multitile/multitile_bump.dm:486]:
```suggestion
			to_chat(V.seats[VEHICLE_DRIVER], SPAN_WARNING(SPAN_BOLD("YOU RAMMED AN ALLY AND HURT THEM!")))
```
idk if the allcaps is 100% needed

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm (hunk 1 above)
Comment (JeserTheFox) [reply code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm]: Fixed.

File: code/modules/vehicles/hardpoints/wheels/locomotion.dm (hunk 2 above)
Comment (JeserTheFox) [reply code/modules/vehicles/hardpoints/wheels/locomotion.dm]: Changed to take_damage.

File: code/modules/vehicles/multitile/multitile_bump.dm (hunk 3 above)
Comment (JeserTheFox) [reply code/modules/vehicles/multitile/multitile_bump.dm:486]: As I found out, DANGER style is considered combat so if you have combat chatlogs disabled, you won't see them, otherwise I'd made it in DANGER style and no all-caps.

**Review by stanalbatross — CHANGES_REQUESTED** [review]
you should make the "exit view" button for the side seats also reset your view

**Review by stanalbatross — APPROVED** [review]

**Comment by Diegoflores31** [comment]
**APC Gunner gets small zoom to increase range it can provide covering fire from.**
tested on TM and literally provides a seizure to whoever is manning the Gunner side of the APC . i would recomend either adjust it to ALL the screen instead of just making it 1 more tile in front or remove it because it just makes it unplayable.

**Comment by JeserTheFox** [comment]
> **APC Gunner gets small zoom to increase range it can provide covering fire from.** tested on TM and literally provides a seizure to whoever is manning the Gunner side of the APC . i would recomend either adjust it to ALL the screen instead of just making it 1 more tile in front or remove it because it just makes it unplayable.

Gotta agree, I forgot that both tank and turret rotation times were longer, unlike APC. And since gunner doesn't control rotation, it ends up jarring as hell, gonna just give a minor view increase without offset.

**Review by stanalbatross — APPROVED** [review]
